Tests/SentryTests/Networking/SentryHttpTransportTests.swift

```diff
@@ -614,6 +619,72 @@ class SentryHttpTransportTests: XCTestCase {
         
         assertEventIsSentAsEnvelope()
     }
+    
+    func testFlush_BlocksCallingThread() {
+        CurrentDate.setCurrentDateProvider(DefaultCurrentDateProvider.sharedInstance())
+        
+        givenCachedEvents()
+        
+        fixture.requestManager.responseDelay = fixture.flushTimeout + 0.2
+        
+        let beforeFlush = Date()
+        let success = sut.flush(timeout: fixture.flushTimeout)
+        let blockingDuration = Date().timeIntervalSince(beforeFlush)
```

Comment:
@bruno-garcia or @mattjohnsonpint, do you have any suggestions on how to properly test if a call to a function is blocking?

Reply:
We have a [similar test in .NET](https://github.com/getsentry/sentry-dotnet/blob/cb735d1f62e062b6cf1ef3e75f22e914f0c98668/test/Sentry.Tests/Internals/BackgroundWorkerTests.cs#L361-L384), where we also just test that the flush timeout period has elapsed.

We do the blocking slightly differently, by using an internal `process: false` flag to the background worker, which just tells it not to process items after enqueueing them.  I believe a previous version of the test took an approach similar to yours where we just added an extra long delay to the transport.  I can't recall why we switched approaches, but I think for this particular case either would work.

Reply:
My only suggestion would be to use a monotonic timing source, but I think you already made that change.

Reply:
Thanks, @mattjohnsonpint.